app/scripts/lib/validate.js

```diff
@@ -146,6 +151,15 @@ define(function (require, exports, module) {
       return _.all(_.keys(data), function (key) {
         return isDataValid(data[key], schema[key]);
       });
+    },
+
+    /**
+     * Check whether string is a url. Validates url with or without http(s) and trailing slash.
+     *
+     * http://blog.mattheworiordan.com/post/13174566389/url-regular-expression-for-links-with-or-without
+     */
+    isUrlValid: function isUrlValid (url){
+      return urlRegEx.test(url);
```

Comment:
It would fail for the `urn`, will update to validate this. When does the `redirectUri` get set in the Sync Relier? Am I looking in right spot?

https://github.com/mozilla/fxa-content-server/blob/7d8c2c0661c4492895d39cc10ebcbd602f4bf596/app/scripts/models/reliers/sync.js#L38